tiledb/sm/array/array.cc

```diff
@@ -864,7 +864,13 @@ Status Array::compute_non_empty_domain() {
     auto metadata_num = fragment_metadata_.size();
     for (size_t j = 1; j < metadata_num; ++j) {
       const auto& meta_dom = fragment_metadata_[j]->non_empty_domain();
-      array_schema_->domain()->expand_ndrange(meta_dom, &non_empty_domain_);
+      if (!meta_dom.empty())
+        array_schema_->domain()->expand_ndrange(meta_dom, &non_empty_domain_);
+      else {
+        LOG_STATUS(Status::ArrayError(
```

Comment:
did you mean `return LOG_STATUS(...`?

Reply:
No, by avoiding the returning and just logging we gracefully skip their fragment in consideration of the non empty domain and it allows the non-empty domain request to proceed.

Reply:
Oh I see, could you please add a comment above to explain?

Reply:
Done